final-paper/final-paper.org

```diff
@@ -61,28 +60,30 @@ Since vectors with the greatest similarity may not satisfy the predicate, this a
+In many cases, it is infeasible to construct an oracle index for every possible predicate, and thus a general-purpose index such as ACORN is more desirable, as it performs reasonably well across a diverse array of queries and is much more space efficient.
 
-However, access patterns are sometimes not uniformly distributed.
-Consider, for example, queries about football teams during Superbowl or states during election night.
-In certain scenarios, a general-purpose predicate-agnostic index can be supplemented by carefully chosen supplemental indices constructed using frequently occurring predicates.
-This is the goal of OAK.
+There are, however, many query distributions could be get significant and meaningful performance improvement if an oracle index were to exist that could be searched rather than an index that also contains vectors that don't match the search predicate.
+The greater the scale of the database and the volume of queries, the more critical such a performance improvement could be. 
+Consider, for example, queries about football teams during the Superbowl, or about states during election night.
+In scenarios like this, a general-purpose predicate-agnostic index can be supplemented by carefully chosen supplemental indices that are constructed so as to effectively serve some frequently occurring search predicates.
+Intelligently routing to supplemental indexes in such cases is the goal of OAK.
 
 * Background and Related Work
 
-The underlying data structure of both ACORN and OAK is Hierarchical Navigable Small Worlds (HNSW)[cite:https://arxiv.org/abs/1603.09320], a hierarchical, tree-link structure where each node of the tree represents a set of vectors.
-More specifically, it is a /proximity graph/, in which two vertices are linked based on proximity.
+The underlying data structure that enables vector similiarity search in both ACORN and OAK is Hierarchical Navigable Small Worlds (HNSW) [cite:@malkovEfficientRobustApproximate2018], an index that makes use of a hierarchical, tree-link structure during search to more quickly work through the plausible search space while retaining reasonable recall.
+More specifically, HNSW leverages a /proximity graph/, in which two vertices are linked based on proximity.
 Proximity is usually computed using Euclidean distance, though other similarity metrics exist (e.g. cosine similarity).
 
 ACORN modifies the HNSW construction algorithm to use neighbor expansion, creating a denser graph.
 While HNSW collects $M$ approximate nearest neighbors as candidate edges for each node in the index, ACORN collects $M \dot \gamma$ approximate nearest neighbors as candidate edges per node.
 The intuition is that given enough redundant nodes, the search space is sufficiently large, even when filtering based on the predicate during search.
```

Comment:
I think we could use a finer grained description of ACORN and its neighbor expansion here. (ACORN can /either/ make a denser graph, ACORN-gamma, or can do the expansion at search time, ACORN-1 -- if I remember correctly.)

Reply:
Added a TODO.